Documentation/raspi-i2c.md

```diff
@@ -0,0 +1,162 @@
+# Enabling I2C on Raspberry Pi
+
+If you want to use the I2C capabilities of your Raspberry Pi, you will need to activate this feature. And if you want to run your code without elevated root privileged to access them, then, you may as well need to make couple of modifications. This tutorial is here to help you activating I2C and making sure you'll get the right permissions.
+
+## Basic Hardware I2C code
+
+The most simplest code you can build to use hardware I2C is the following (require C#9.0 to be activated):
+
+```csharp
+using System;
+using System.Device.I2c;
+
+Console.WriteLine("Hello I2C!");
+I2cDevice i2c = I2cDevice.Create(new I2cConnectionSettings(1, 0x12));
+i2c.WriteByte(0x42);
+var read = i2c.ReadByte();
+```
+
+In this case ```I2cDevice.Create(new I2cConnectionSettings(1, 0x12))``` will create an I2C device on the bus 1 (the default one) and with the device address 0x12.
+
+If you get an error message like this one, it means I2C has not been properly enabled:
+
+```text
+Hello I2C!
+Unhandled exception. System.IO.IOException: Error 2. Can not open I2C device file '/dev/i2c-1'.
+   at System.Device.I2c.UnixI2cDevice.Initialize()
+   at System.Device.I2c.UnixI2cDevice.WriteByte(Byte value)
+   at TestTest.Program.Main(String[] args) in C:\tmp\TestI2C\Program.cs:line 5
+Aborted
+```
+
+If you get an error message like the following one, it means that most likely you have a problem with cabling or your I2C address:
+
+```text
+Unhandled exception. System.IO.IOException: Error 121 performing I2C data transfer.
+   at System.Device.I2c.UnixI2cDevice.ReadWriteInterfaceTransfer(Byte* writeBuffer, Byte* readBuffer, Int32 writeBufferLength, Int32 readBufferLength)
+   at System.Device.I2c.UnixI2cDevice.Transfer(Byte* writeBuffer, Byte* readBuffer, Int32 writeBufferLength, Int32 readBufferLength)
+   at System.Device.I2c.UnixI2cDevice.WriteByte(Byte value)
+   at TestTest.Program.Main(String[] args) in C:\tmp\TestI2C\Program.cs:line 14
+Aborted
+```
+
```

Comment:
Add: "In rare cases, you might see the above exception during normal operation of a device as well. Adding retries might solve the issue."

Reply:
Yes, good point. It does not exist on all systems. But right, it does on some! Will add!